docs/sessions.md

```diff
@@ -264,6 +264,40 @@ sess.sql("SELECT * FROM example.tbl, temp LIMIT 1").show()
 
     We aim to support SQL DDL in future releases!
 
+### Using providers
+
+Various AI/ML protocols can be implemented through Providers. Here is an example of using OpenAI provider for creating text embeddings
+
+```python
+# Set the current provider to be OpenAI (requires a valid OpenAI API key)
+api_key = <your OpenAI API key here>
+sess.set_provider("openai", api_key=api_key)
+
+# Check the current session provider
+print(sess.current_provider().name)
+"""
+openai
+"""
+
+# Get the default text embedder model used by the provider
+default_model = sess.current_provider().get_text_embedder()
+print(default_model.model_name)
+"""
+text-embedding-3-small
+"""
+
+# Use the default embedder model to create embeddings
+result = default_model.instantiate().embed_text(["Hello world"])
+
+# Use a custom OpenAI embedder model to create embeddings
+result_large = (
+    sess.current_provider()
+    .get_text_embedder(model="text-embedding-3-large")
+    .instantiate()
+    .embed_text(["Hello world"])
+)
+```
```

Comment:
For these examples, it would be nice not to put in sessions.md, but in the docs examples directory in a longer form embeddings doc.